python/pyspark/sql/readwriter.py

```diff
@@ -328,6 +328,10 @@ def csv(self, path, schema=None, sep=None, encoding=None, quote=None, escape=Non
                            applies to both date type and timestamp type. By default, it is None
                            which means trying to parse times and date by
                            ``java.sql.Timestamp.valueOf()`` and ``java.sql.Date.valueOf()``.
+        :param timezone: defines the timezone to be used for both date type and timestamp type.
+                         If a timezone is specified in the data, this will load them after
```

Comment:
I mean specifically that any "timezone" specified as a parameter when reading the input "27/08/2015 00:00 PDT" should not matter, which is why I wonder why this is in the example. The parsed timestamp is unambiguously 1440658800000; it has no timezone because that's a concept only related to formatted dates/times.

When outputting formatted strings, I understand why a timezone parameter matters. For example with format "dd/MM/yyyy HH:mm:ss z" the output depends on the desired timezone. With "GMT", yes I agree with your new example, the output is "27/08/2015 07:00:00 GMT"

Back to the original question -- what is the comment referring to then? I don't see a need to manually adjust for a timezone.